source/device/client.py

```diff
@@ -7,10 +7,11 @@ def __init__(self, address="192.168.0.69", port=50010, autoconnect=True, timeout
         self._address = address
         self._port = port
         self._autoconnect = autoconnect
+        self._device_timeout = timeout
         self._rpc = None
         if autoconnect:
-            self._rpc = O2x5xxRPCDevice(address=self._address)
-        super(O2x5xxPCICDevice, self).__init__(address, port, autoconnect, timeout)
+            self._rpc = O2x5xxRPCDevice(address=self._address, timeout=self._device_timeout)
```

Comment:
@Galoshi What do you think about that?

Reply:
Keeping older customer code backwards compatible i would leave the autoconnect flag as True.
One of our customers asked for connect and disconnect methods for the PCIC client to receiving full control over it.

The PCIC client in the rpc/client.py is needed for reading the answer after the trigger. The trigger method in RPC is not waiting for response of the device. It may happen that if the sensor needs about 1 sec for evaluation that you will read the answer n-1.

Reply:
It's not about the flag. I am asking about if line 12 `if autoconnect`. Why is it needed? Can't we just create rpc and pcic object and don't bother with having self.__timeout nor having checks in `__enter__`/@parameter?

RPC have to connect to anything in `__init__`?

I will add issue about this PCIC client in `rpc/client.py` Can't see any pros of that. If you want trigger output use PCIC.

Reply:
@cfreundl Yes thats right. Customer should use either the O2x5xxPCICDevice, O2x5xxRPCDevice or O2x5xxDeviceV2.
O2x5xxDevice was added for keeping "older" customer code compatible.

@bnozka I agree. I will delete the autoconnect check in devices and pass them to the PCIC client which is fine.

Reply:
fixed in a75903d694aab696956472159ad3c0284799b000